src/fs_event.c

```diff
@@ -0,0 +1,103 @@
+#include "entry.h"
+
+static int fs_event_method_get(void *p, Janet key, Janet *out);
+static const JanetAbstractType fs_event_type = {
+    "uv/fs-event",
+    NULL,
+    juv_handle_mark,
+    fs_event_method_get,
+    JANET_ATEND_GET
+};
+
+static Janet cfun_fs_event_new(int32_t argc, Janet *argv) {
+    janet_fixarity(argc, 1);
+    JanetFiber *fiber = janet_getfiber(argv, 0);
+    uv_fs_event_t* handle = (uv_fs_event_t*) juv_makehandle(&fs_event_type, sizeof(*handle));
+    int ret = uv_fs_event_init(uv_default_loop(), handle);
+    if (ret < 0) janet_panic("could not create fs event monitor");
+    Janet val = juv_wrap_handle(handle);
+    janet_gcroot(val);
+    juv_handle_setfiber(handle, 0, fiber);
+    return val;
+}
+
+static void fs_event_cleanup(uv_fs_event_t *t) {
+    int res = uv_fs_event_stop(t);
+    janet_gcunroot(juv_wrap_handle(t));
+    juv_handle_setfiber(t, 0, NULL);
+    juv_handle_setcb(t, 1, NULL);
+    if (res < 0) juv_panic(res);
+}
+
+static void juv_fs_event_cb(uv_fs_event_t *handle, const char *filename, int events, int status) {
+    if (status) {
+        juv_panic(status);
+    } else {
+        JanetFunction *cb = juv_handle_cb(handle, 1);
+        Janet argv[] = { juv_wrap_handle(handle),
+                         janet_wrap_string(janet_cstring(filename)),
+                         janet_wrap_integer(events) };
+        Janet out = janet_wrap_nil();
+        JanetFiber *cb_f = janet_fiber(cb, 64, 3, argv);
+        JanetSignal sig;
+        if (!cb_f) {
+            out = janet_cstringv("arity mismatch");
+            sig = JANET_SIGNAL_ERROR;
+        } else {
+            JanetFiber *handle_f = juv_handle_fiber(handle, 0);
+            cb_f->env = handle_f->env;
+            sig = janet_continue(cb_f, janet_wrap_nil(), &out);
+        }
+        if (sig != JANET_SIGNAL_OK && sig != JANET_SIGNAL_YIELD) {
+            janet_stacktrace(cb_f, out);
```

Comment:
When testing this, I found it helpful to have a stacktrace output in addition to the error message.